workspace.rkt

```diff
@@ -11,6 +20,23 @@
 
 (define workspace-folders (mutable-set))
 
+(define (didRenameFiles params)
+  (match-define (RenameFilesParams #:files files) params)
+  (for ([f files])
+    (match-define (FileRename #:oldUri old-uri #:newUri new-uri) f)
+
+    ; remove all awaiting internal queries about `old-uri`
+    (clear-old-queries/doc-close old-uri)
+
+    (if (string-suffix? new-uri ".rkt")
+      (let ([safe-doc (hash-ref open-docs (string->symbol old-uri) #f)])
+        ; `safe-doc = #f` should be rarely happened.
+        ; we simply give up to handle it, let's trust LSP client will send others request about analysis this file.
+        (when safe-doc
+          (doc-update-uri! safe-doc new-uri)
```

Comment:
`safe-doc` is a `SafeDoc` struct contains a `Doc` struct and a rwlock, the `Doc` struct stores its information. So we need to acquire its lock like this:

```racket
(with-write-doc safe-doc
  (lambda (doc)
    (doc-update-uri! doc new-uri)))
```